rules/private/phases/phase_semanticdb.bzl

```diff
@@ -8,9 +8,9 @@ load("@rules_scala_annex//rules:providers.bzl", _ScalaConfiguration = "ScalaConf
 # SemanticDB compiler plugin being enabled.
 #
 def phase_semanticdb(ctx, g):
-    scala_configuration = ctx.attr.scala[_ScalaConfiguration]
+    toolchain = ctx.toolchains["//rules/scala:toolchain_type"]
```

Comment:
I imagine this string will never change, but it may be worth doing what the `rules_go` folks did and instead use a constant that we import. That way if this string ever needs to change, it will be easy to change.

https://github.com/bazel-contrib/rules_go/blob/a54de7c88bb3cb133f951f0e9817c09be42056cb/go/private/common.bzl#L16

Also, they use the fully qualified name, which is probably  not a bad idea either. So `@rules_scala_annex//rules/scala:toolchain_type`

Reply:
This sounds like a great idea, but if we ever need to move that constant, we'll need to update all the imports that reference it, which introduces the same problem.

Also, Is the fully-qualified name unnecessary? I prefer not to use those unless it's necessary. My understanding was that Bazel understood this was in `rules_scala_annex`, even if these rules are being used in another repository.

Reply:
Oh, you meant defining that alias for internal use in this file. That makes a lot of sense; I'll do that.